### PR TITLE
Add proper following support

### PR DESCRIPTION
**Pull request by odinuge**
With fancy optimistic loading, since it looks shitty without it...

### CODE REVIEW

File: app/reducers/index.js [hunk 1]
```diff
@@ -229,3 +183,93 @@ export const surveySubmissionSchema = new schema.Entity('surveySubmissions', {
...
+);
+
+export const [
+  followersCompanySchema,
+  followersCompany,
+] = followersSchemaGenerator('Company');
+
+export const [followersUserSchema, followersUser] = followersSchemaGenerator(
+  'User'
+);
```
Comment (LudvigHz) [inline app/reducers/index.js:222]: Maybe a bit random to have this here? I think it's a bit cleaner to have this with the followers selectors.

File: app/reducers/index.js [hunk 2]
```diff
@@ -229,3 +183,93 @@ export const surveySubmissionSchema = new schema.Entity('surveySubmissions', {
...
+) => {
+  const key = `followers${title}`;
+  return [
+    new schema.Entity(key, {
+      follower: userSchema,
+    }),
+    createEntityReducer({
+      key,
+      types,
+    }),
```
Comment (LudvigHz) [inline app/reducers/index.js:203]: How does mutating the follower states work now? I'm not familiar with the default handling in `createEntityReducer` :upside_down_face:.

**Review by LudvigHz — COMMENTED** [review]
Awesome! It's a bit complicated, but I'm assuming it makes sense.

Is there any reason to have a separate schema for each target on the followers. The only difference I can see is the target type, so why not just have the target type on the schema and have all the followers in one state? (Sort of like comments)

File: app/reducers/index.js (hunk 2 above)
Comment (odinuge) [reply app/reducers/index.js:203]: It woks as expected (normal CRUD) as long as the actions define the correct schema.

File: app/reducers/index.js (hunk 1 above)
Comment (odinuge) [reply app/reducers/index.js:222]: Guess we can do that, but then we end with twice as much code, and need to defined the various following types twice (bigger change for errors), but guess we can move them.

**Review by LudvigHz — COMMENTED** [review]
This looks great now! I think a couple of tests is all that's missing before we can get this merged :rocket:

**Pull request merged**